pkg/apis/awsprovider/v1beta1/register.go

```diff
@@ -43,81 +43,66 @@ var (
 	SchemeBuilder = &scheme.Builder{GroupVersion: SchemeGroupVersion}
 )
 
-// AWSProviderConfigCodec is a runtime codec for the provider configuration
-// +k8s:deepcopy-gen=false
-type AWSProviderConfigCodec struct {
-	encoder runtime.Encoder
-	decoder runtime.Decoder
-}
-
-// NewScheme creates a new Scheme
-func NewScheme() (*runtime.Scheme, error) {
-	return SchemeBuilder.Build()
-}
-
-// NewCodec creates a serializer/deserializer for the provider configuration
-func NewCodec() (*AWSProviderConfigCodec, error) {
-	scheme, err := NewScheme()
-	if err != nil {
-		return nil, err
+// RawExtensionFromProviderSpec marshals the machine provider spec.
```

Comment:
Do we have any tests for any of the stuff in this file?

Reply:
No, I'm planning to add the in a follow-up. We also need them for vsphere